### PR TITLE
Generalise runtime metadata

### PR DESCRIPTION
**Pull request by challurip**
Generalised runtime metadata and able to use them in ~~failed~~ event handlers.

~Close~ Address main part of #2330
Still to do https://github.com/cylc/cylc/issues/2330#issuecomment-316733946

### CODE REVIEW

File: lib/cylc/network/https/base_client.py [hunk 1]
```diff
@@ -265,7 +264,7 @@ def _get_data_from_url_with_urllib2(self, http_request_items):
                 else:
                     raise ConnectionError(url, exc)
             except Exception as exc:
-                if cylc.flags.debug:
+                if not cylc.flags.debug:
```
Comment (matthewrmshin) [inline lib/cylc/network/https/base_client.py]: Changes in this file are probably unrelated?

File: doc/src/cylc-user-guide/suiterc.tex [hunk 2]
```diff
@@ -1280,6 +1216,71 @@ \subsection{[runtime]}
...
+\begin{myitemize}
+\item {\em type:} string (URL)
+\item {\em default:} (none)
+\item {\em example:} you can set URLs to all tasks in a suite by putting
+    something like the following in the root namespace:
+    \begin{lstlisting}
+[runtime]
+    [[root]]
+        URL = http://suites/${CYLC_SUITE_NAME}/${CYLC_TASK_NAME}.html
+    \end{lstlisting}
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/suiterc.tex]: Section level needs updating?

File: doc/src/cylc-user-guide/suiterc.tex [hunk 3]
```diff
@@ -1280,6 +1216,71 @@ \subsection{[runtime]}
...
+Replace \_\_MANY\_\_ with namespace name and value, and repeat as needed to define all the user-defined metadata items of the task.
+Names may contain letters, digits, underscores and hyphens and values may contain letters and digits.
+
+\begin{myitemize}
+\item {\em type:} String or integer
+\item {\em default:} (none)
+\item {\em example:}
+    \begin{lstlisting}
+importance = high
+color = red
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/suiterc.tex]: Should include sections in example?

**Comment by matthewrmshin** [comment]
Looks like this is almost there. Just 2 minor Travis CI test failures, and perhaps a bit more docs?

File: lib/cylc/network/https/base_client.py (hunk 1 above)
Comment (hjoliver) [inline lib/cylc/network/https/base_client.py]: (the `not` seems to be a mistake?)

**Comment by matthewrmshin** [comment]
Note PEP8 test failure on Travis CI.

```
/home/travis/build/cylc/cylc/lib/cylc/config.py:706:21: E128 continuation line under-indented for visual indent

/home/travis/build/cylc/cylc/lib/cylc/config.py:708:21: E128 continuation line under-indented for visual indent
```

**Pull request merged**